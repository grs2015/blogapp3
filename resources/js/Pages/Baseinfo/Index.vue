<template>
    <div>Baseinfo</div>
    <div>{{ model.baseinfo.address }}</div>
    <div>{{ model.baseinfo.phone }}</div>
    <div>{{ model.baseinfo.email }}</div>
    <div>{{ model.baseinfo.website }}</div>
    <div>{{ model.baseinfo.content }}</div>
    <div>{{ model.baseinfo.full_address }}</div>
</template>

<script setup>
defineProps({ model: Object })

</script>
