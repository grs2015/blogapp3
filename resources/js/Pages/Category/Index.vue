<template>

</template>

<script setup>
defineProps({ model: Object })
</script>
