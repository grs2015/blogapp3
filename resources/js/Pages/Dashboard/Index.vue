<script setup>

import { Head } from '@inertiajs/inertia-vue3'

</script>

<template>
    <div>
        Page content
    </div>
</template>
