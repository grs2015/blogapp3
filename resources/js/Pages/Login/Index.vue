<script lang="ts">
export default {
    layout: Layout
}
</script>

<script setup lang="ts">

import Layout from '@/Shared/RegisterLayout.vue'
import { useForm, Link } from '@inertiajs/inertia-vue3';
import { ref } from 'vue'

const isPassword = ref(true)
const form = useForm({
    email: null,
    password: null
})

const loginUser = () => form.post('/login')

</script>

<template>
    <div class="row flex-center fit window-height">
        <q-form @submit.prevent="loginUser" style="width: 30%">
            <q-card flat bordered>
                <q-card-section class="text-primary text-h6">
                    Register Form
                </q-card-section>
                <q-card-section>
                    <div class="column q-col-gutter-y-md">
                        <div class="col">
                            <q-input v-model="form.email" dense clearable clear-icon="close" outlined label="Your email"
                                type="text" bottom-slots
                                :error-message="form.errors.email"
                                :error="!!(form.errors.email)">
                                <template v-slot:prepend>
                                    <q-icon name="email" color="orange" />
                                </template>
                            </q-input>
                        </div>
                        <div class="col">
                            <q-input v-model="form.password" dense clearable clear-icon="close" outlined label="Your password"
                                :type="isPassword ? 'password' : 'text'" bottom-slots
                                :error-message="form.errors.password"
                                :error="!!(form.errors.password)">
                                <template v-slot:prepend>
                                    <q-icon name="lock" color="orange" />
                                </template>
                                <template v-slot:append>
                                    <q-icon
                                        :name="isPassword ? 'visibility_off' : 'visibility'"
                                        class="cursor-pointer"
                                        @click="isPassword = !isPassword"
                                    />
                                </template>
                            </q-input>
                        </div>
                    </div>
                </q-card-section>
                <q-card-actions align="right" class="q-gutter-sm">
                    <Link as="div" href="/">
                        <q-btn flat color="secondary" :disable="form.processing">Cancel</q-btn>
                    </Link>
                    <q-btn flat color="primary" type="submit" :loading="form.processing">Login</q-btn>
                </q-card-actions>
            </q-card>
        </q-form>
    </div>
</template>
