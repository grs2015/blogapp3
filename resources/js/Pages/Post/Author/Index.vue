<script lang="ts">

import Layout from '@/Shared/LayoutAuthor.vue'

export default {
    layout: Layout,
}

</script>

<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import PostTableAuthor from '@/Shared/Tables/PostAuthorTable.vue'
import { postsAuthorBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { Paginated } from '@/Interfaces/PaginatedData'
import { SortingData, breadcrumbsData } from '@/Interfaces/PaginatedData'

interface Props {
    model: {
        posts: Paginated,
        sorting: SortingData
    }
}

const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(postsAuthorBreadcrumbs)

</script>

<template>
    <Head title="Blog posts" />
    <Breadcrumbs :data="breadcrumbs" />
    <PostTableAuthor :paginatedData="props.model.posts" :sortingData="props.model.sorting" />
</template>
