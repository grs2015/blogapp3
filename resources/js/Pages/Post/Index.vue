<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import PostTable from '@/Shared/Tables/PostTable.vue'
import { postsBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { Paginated, SortingData, breadcrumbsData } from '@/Interfaces/PaginatedData'

interface Props {
    model: {
        posts: Paginated,
        sorting: SortingData
    }
}

const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(postsBreadcrumbs)

</script>

<template>
    <Head title="Blog posts" />
    <Breadcrumbs :data="breadcrumbs" />
    <PostTable :paginatedData="props.model.posts" :sortingData="props.model.sorting"/>
</template>
