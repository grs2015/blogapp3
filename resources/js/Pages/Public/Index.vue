<script lang="ts">
    export default {
        layout: Layout

    // layout: (h, page) => h(Layout, () => child),
}
</script>

<script setup lang="ts">

import Layout from '@/Shared/PublicLayout.vue'
import { Head } from '@inertiajs/inertia-vue3'
import { Paginated } from '@/Interfaces/PaginatedData'
import PublishedPosts from '@/Shared/Lists/PublishedPosts.vue'
import { Inertia } from '@inertiajs/inertia';
import { computed } from 'vue'

interface Props {
    model: {
        posts: Paginated,
    }
}

const props = defineProps<Props>()

const data = computed(() => props.model.posts)
const currentPage = computed(() => props.model.posts.current_page)
// const min = computed(() => props.model.posts.from)
const max = computed(() => props.model.posts.last_page)

const paginate = (value) => { Inertia.get('/', { per_page: 10, page: value }) }
</script>

<template>
    <Head title="Actual posts" />

        <div class="q-mt-xl">
            <PublishedPosts :paginatedData="data" />
            <div class="q-pa-lg flex flex-center">
                <q-pagination
                    v-model="currentPage"
                    :max=max
                    @update:model-value="paginate"
                    unelevated
                    color="teal" />
            </div>
        </div>

</template>


