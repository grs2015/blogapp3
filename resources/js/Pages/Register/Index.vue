<script lang="ts">
export default {
    layout: Layout
}
</script>

<script setup lang="ts">

import Layout from '@/Shared/RegisterLayout.vue'
import { useForm, Link } from '@inertiajs/inertia-vue3';
import { UserRole } from '@/Interfaces/PaginatedData'
import { ref } from 'vue'

const isPassword = ref(true)
const form = useForm({
    first_name: null,
    last_name: null,
    email: null,
    password: null,
    password_confirmation: null,
    role: UserRole.Author
})

const registerUser = () => form.post('/register')

</script>

<template>
    <div class="row flex-center fit window-height">
        <q-form @submit.prevent="registerUser" style="width: 30%">
            <q-card flat bordered>
                <q-card-section class="text-primary text-h6">
                    Register Form
                </q-card-section>
                <q-card-section>
                    <div class="column q-col-gutter-y-md">
                        <div class="col">
                            <q-input v-model="form.first_name" dense clearable clear-icon="close" outlined label="Your first name"
                                type="text" bottom-slots
                                :error-message="form.errors.first_name"
                                :error="!!(form.errors.first_name)">
                                <template v-slot:prepend>
                                    <q-icon name="person" color="orange" />
                                </template>
                            </q-input>
                        </div>
                        <div class="col">
                            <q-input v-model="form.last_name" dense clearable clear-icon="close" outlined label="Your last name"
                                type="text" bottom-slots
                                :error-message="form.errors.last_name"
                                :error="!!(form.errors.last_name)">
                                <template v-slot:prepend>
                                    <q-icon name="person" color="orange" />
                                </template>
                            </q-input>
                        </div>
                        <div class="col">
                            <q-input v-model="form.email" dense clearable clear-icon="close" outlined label="Your email"
                                type="email" bottom-slots
                                :error-message="form.errors.email"
                                :error="!!(form.errors.email)">
                                <template v-slot:prepend>
                                    <q-icon name="email" color="orange" />
                                </template>
                            </q-input>
                        </div>
                        <div class="col">
                            <q-input v-model="form.password" dense clearable clear-icon="close" outlined label="Your password"
                                :type="isPassword ? 'password' : 'text'" bottom-slots
                                :error-message="form.errors.password"
                                :error="!!(form.errors.password)">
                                <template v-slot:prepend>
                                    <q-icon name="lock" color="orange" />
                                </template>
                                <template v-slot:append>
                                    <q-icon
                                        :name="isPassword ? 'visibility_off' : 'visibility'"
                                        class="cursor-pointer"
                                        @click="isPassword = !isPassword"
                                    />
                                </template>
                            </q-input>
                        </div>
                        <div class="col">
                            <q-input v-model="form.password_confirmation" dense clearable clear-icon="close" outlined label="Confirm your password"
                                :type="isPassword ? 'password' : 'text'" bottom-slots
                                :error-message="form.errors.password_confirmation"
                                :error="!!(form.errors.password_confirmation)">
                                <template v-slot:prepend>
                                    <q-icon name="lock" color="orange" />
                                </template>
                                <template v-slot:append>
                                    <q-icon
                                        :name="isPassword ? 'visibility_off' : 'visibility'"
                                        class="cursor-pointer"
                                        @click="isPassword = !isPassword"
                                    />
                                </template>
                            </q-input>
                        </div>
                        <div class="col">
                            <q-radio v-model="form.role" val="author" label="Author" />
                            <q-radio v-model="form.role" val="member" label="Commentor" />
                        </div>
                    </div>
                </q-card-section>
                <q-card-actions align="right" class="q-gutter-sm">
                    <Link as="div" href="/">
                        <q-btn flat color="secondary" :disable="form.processing">Cancel</q-btn>
                    </Link>
                    <q-btn flat color="primary" type="submit" :loading="form.processing">Register</q-btn>
                </q-card-actions>
            </q-card>
        </q-form>
    </div>
</template>
