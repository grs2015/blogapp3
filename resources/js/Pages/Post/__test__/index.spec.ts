import { mount } from '@vue/test-utils'
import Index from '@/Pages/Post/Index.vue'
import { Head } from '@inertiajs/inertia-vue3'

describe('First test', () => {
    test('Nothing to test', () => {
        const wrapper = mount(Index, { props: { model: { posts: { data: [] } } }, shallow: true })
        expect(wrapper.findComponent(Head).exists()).toBe(true)
        expect(wrapper.findComponent(Head).attributes('title')).toBe('Blog posts')
        expect(true).toBe(true)

    })
})
