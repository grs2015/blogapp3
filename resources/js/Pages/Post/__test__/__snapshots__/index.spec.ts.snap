// Vitest Snapshot v1

exports[`Post/Index component > makes a snapshot 1`] = `
<div
  data-v-app=""
>
  
  <head-stub
    title="Blog posts"
  />
  <breadcrumbs-stub
    data="[object Object],[object Object]"
  />
  <post-table-stub
    paginateddata="[object Object]"
  />
  
</div>
`;
